rs/boundary_node/rate_limits/api/src/schema_versions/v1.rs

```diff
@@ -209,13 +296,67 @@ mod test {
                 subnet_id: None,
                 methods_regex: Some(Regex::new("^.*$").unwrap()),
                 request_types: None,
+                ip: Some(IpNet::new_assert(
+                    IpAddr::V4(Ipv4Addr::new(10, 1, 1, 0)),
+                    24
+                )),
+                ip_prefix_group: Some(IpPrefixes { v4: 24, v6: 64 }),
                 limit: Action::Limit(100, Duration::from_secs(1)),
             }
         );
 
+        // Bad prefix lengths
+        let rule_raw = indoc! {"
+        canister_id: aaaaa-aa
+        methods_regex: ^.*$
+        ip: 10.1.1.0/24
+        ip_prefix_group:
+          v4: 33
+          v6: 64
+        limit: 100/1s
+        "};
+
+        assert!(RateLimitRule::from_bytes_yaml(rule_raw.as_bytes()).is_err());
```

Comment:
Suggestion: consider asserting error type too, it should not be complex

Reply:
Well, error types here are mostly all `de::Error` so it's hard to assert. I've added some string matches, which isn't nice but should be ok for tests.